File: app/src/main/java/soup/movie/di/ui/MainUiModule.java
package soup.movie.di.ui;

import dagger.Module;
import dagger.Provides;
import soup.movie.di.scope.ActivityScope;
import soup.movie.ui.main.MainContract;
import soup.movie.ui.main.MainPresenter;

@Module
public class MainUiModule {

    @ActivityScope
    @Provides
    MainContract.Presenter presenter() {
        return new MainPresenter();
    }
}

File: app/src/main/java/soup/movie/di/ui/MainUiModule.kt
package soup.movie.di.ui

import androidx.lifecycle.ViewModel
import dagger.Binds
import dagger.Module
import dagger.multibindings.IntoMap
import soup.movie.di.key.ViewModelKey
import soup.movie.ui.main.MainViewModel

@Module
abstract class MainUiModule {

    @Binds
    @IntoMap
    @ViewModelKey(MainViewModel::class)
    abstract fun bindMainViewModel(viewModel: MainViewModel): ViewModel
}
